drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/GuidedDecisionTableModellerViewImpl.java

```diff
@@ -71,613 +66,671 @@
-                Label patternLabel = makePatternLabel( p );
-                patternHeaderPanel.add( patternLabel );
-                patternPanel.add( patternHeaderPanel );
-                patternPanel.add( conditionsPanel );
-                patternsPanel.add( patternPanel );
+                patternHeaderPanel.setStylePrimaryName(GuidedDecisionTableResources.INSTANCE.css().patternSectionHeader());
+                Label patternLabel = makePatternLabel(p);
+                patternHeaderPanel.add(patternLabel);
+                patternPanel.add(patternHeaderPanel);
+                patternPanel.add(conditionsPanel);
+                patternsPanel.add(patternPanel);
 
                 List<ConditionCol52> conditions = p.getChildColumns();
-                for ( ConditionCol52 c : conditions ) {
+                for (ConditionCol52 c : conditions) {
                     HorizontalPanel hp = new HorizontalPanel();
-                    hp.setStylePrimaryName( GuidedDecisionTableResources.INSTANCE.css().patternConditionSectionHeader() );
-                    if ( isEditable ) {
-                        hp.add( removeCondition( c ) );
-                    }
-                    hp.add( editCondition( p,
-                                           c ) );
-                    SmallLabel conditionLabel = makeColumnLabel( c );
-                    hp.add( conditionLabel );
-                    conditionsPanel.add( hp );
+
+                    SmallLabel conditionLabel = makeColumnLabel(c);
+                    hp.add(conditionLabel);
+
+                    final FlowPanel buttons = new FlowPanel() {{
+                        add(editCondition(p,
+                                          c));
+
+                        if (isEditable) {
+                            add(removeCondition(c));
+                        }
+                    }};
+
+                    hp.add(buttons);
+
+                    conditionsPanel.add(hp);
                 }
-            } else if ( conditionColumn instanceof BRLConditionColumn ) {
+            } else if (conditionColumn instanceof BRLConditionColumn) {
                 BRLConditionColumn brl = (BRLConditionColumn) conditionColumn;
 
                 HorizontalPanel patternHeaderPanel = new HorizontalPanel();
-                patternHeaderPanel.setStylePrimaryName( GuidedDecisionTableResources.INSTANCE.css().patternSectionHeader() );
+                patternHeaderPanel.setStylePrimaryName(GuidedDecisionTableResources.INSTANCE.css().patternSectionHeader());
                 HorizontalPanel patternPanel = new HorizontalPanel();
-                if ( isEditable ) {
-                    patternPanel.add( removeCondition( brl ) );
+
+                Label patternLabel = makePatternLabel(brl);
+                patternPanel.add(patternLabel);
+                patternHeaderPanel.add(patternPanel);
+
+                if (isEditable) {
```

Comment:
Please remove this if. 
![screenshot from 2017-04-20 10-28-54](https://cloud.githubusercontent.com/assets/8044780/25221367/2f7357e2-25b5-11e7-87b7-e8787d63cf5a.png)